### ISSUE TITLE
image adjust

### ISSUE THREAD

**Issue by ideptor** [ideptor/pytorch-resnet18#4]
diff --git a/main.py b/main.py
index 34bda90..4c59277 100644
--- a/main.py
+++ b/main.py
@@ -18,7 +18,7 @@ from time import time
 
 from models import *
 from tqdm import tqdm
-# from utils import progress_bar
+from utils import progress_bar
 
 
 
@@ -48,21 +48,21 @@ def train(epoch):
         total += targets.size(0)
         correct += predicted.eq(targets).sum().item()
 
-        # progress_bar(batch_idx, len(trainloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
-        #              % (train_loss/(batch_idx+1), 100.*correct/total, correct, total))
+        progress_bar(batch_idx, len(trainloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
+                     % (train_loss/(batch_idx+1), 100.*correct/total, correct, total))
 
         # pbar.set_postfix(loss=(train_loss/(batch_idx+1)), acc=(100.*correct/total), hit=f"{correct}/{total}")
-        end = time()
+    end = time()
 
-        log = (
-            f"[{epoch+1}/{end_epoch}][TRAIN {batch_idx+1:02d}/65] "
-            f"Step: {int(end-step_start):d}s{int(((end-step_start)-int(end-step_start))*1000):03d}ms | "
-            f"Tot: {int((end-tot_start)/60):02d}m{int(end-tot_start)%60:02d}s | "
-            f"Loss: {(train_loss/(batch_idx+1)):.6f} | Acc: {(100.*correct/total):.2f}% | Hit: {correct}/{total}"
-        )
-        print(log)
-        logfile_writer.write(log+"\n")
-        logfile_writer.flush()
+    log = (
+        f"[{epoch+1}/{end_epoch}][TRAIN {batch_idx+1:02d}/65] "
+        f"Step: {int(end-step_start):d}s{int(((end-step_start)-int(end-step_start))*1000):03d}ms | "
+        f"Tot: {int((end-tot_start)/60):02d}m{int(end-tot_start)%60:02d}s | "
+        f"Loss: {(train_loss/(batch_idx+1)):.6f} | Acc: {(100.*correct/total):.2f}% | Hit: {correct}/{total}"
+    )
+    # print(log)
+    logfile_writer.write(log+"\n")
+    logfile_writer.flush()
         
 def test(epoch):
     global best_acc
@@ -84,20 +84,20 @@ def test(epoch):
             total += targets.size(0)
             correct += predicted.eq(targets).sum().item()
 
-            # progress_bar(batch_idx, len(testloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
-            #              % (test_loss/(batch_idx+1), 100.*correct/total, correct, total))
+            progress_bar(batch_idx, len(testloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
+                         % (test_loss/(batch_idx+1), 100.*correct/total, correct, total))
             # pbar.set_postfix(loss=(test_loss/(batch_idx+1)), acc=(100.*correct/total), hit=f"{correct}/{total}")
             # print(f"{batch_idx}/4, loss={(test_loss/(batch_idx+1)):.5f}, acc={(100.*correct/total)}({correct}/{total})")
-            end = time()
-            log = (
-                f"[{epoch+1}/{end_epoch}][TEST {batch_idx+1:d}/4] "
-                f"Step: {int(end-step_start):d}s{int(((end-step_start)-int(end-step_start))*1000):03d}ms | "
-                f"Tot: {int((end-tot_start)/60):02d}m{int(end-tot_start)%60:02d}s | "
-                f"Loss: {(test_loss/(batch_idx+1)):.6f} | Acc: {(100.*correct/total):.2f}% | Hit: {correct}/{total}"
-            )
-            print(log)
-            logfile_writer.write(log+"\n")
-            logfile_writer.flush()
+        end = time()
+        log = (
+            f"[{epoch+1}/{end_epoch}][TEST {batch_idx+1:d}/4] "
+            f"Step: {int(end-step_start):d}s{int(((end-step_start)-int(end-step_start))*1000):03d}ms | "
+            f"Tot: {int((end-tot_start)/60):02d}m{int(end-tot_start)%60:02d}s | "
+            f"Loss: {(test_loss/(batch_idx+1)):.6f} | Acc: {(100.*correct/total):.2f}% | Hit: {correct}/{total}"
+        )
+        # print(log)
+        logfile_writer.write(log+"\n")
+        logfile_writer.flush()
 
     # Save checkpoint.
     acc = 100.*correct/total
@@ -116,9 +116,13 @@ def test(epoch):
 
 def get_args() -> dict:
     parser = argparse.ArgumentParser(description='PyTorch Resnet18 Training')
+    parser.add_argument('--epoch', '-e', default=15, type=int, help='epochs')
+    parser.add_argument('--size', '-s', default=128, type=int, help='image size')
+    parser.add_argument('--dataset', '-d', type=str, default="dataset", help='dataset folder')
     parser.add_argument('--lr', default=0.01, type=float, help='learning rate')
     parser.add_argument('--resume', '-r', action='store_true',
                         help='resume from checkpoint')
+    parser.add_argument("--gpu", "-g", action="store_true")
     args = parser.parse_args()
 
     return args
@@ -128,16 +132,20 @@ if __name__ == '__main__':
 
     args = get_args()
 
-    
+    os.makedirs("logs", exist_ok=True)
 
-    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    if args.gpu:
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    else:
+        device = 'cpu'
+    
     best_acc = 0  # best test accuracy
     start_epoch = 0  # start from epoch 0 or last checkpoint epoch
 
     # Data
     print('==> Preparing data..')
     transform_train = transforms.Compose([
-        transforms.RandomCrop(64, padding=4),
+        # transforms.RandomCrop(64, padding=4),
         # transforms.RandomHorizontalFlip(),
         transforms.ToTensor(),
         transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
@@ -151,13 +159,13 @@ if __name__ == '__main__':
 
     # trainset = torchvision.datasets.CIFAR10(
     #     root='./data', train=True, download=True, transform=transform_train)
-    trainset = ImageFolder("dataset/train", transform=transform_train)
+    trainset = ImageFolder(f"{args.dataset}/train", transform=transform_train)
     trainloader = torch.utils.data.DataLoader(
         trainset, batch_size=16, shuffle=True, num_workers=2)
 
     # testset = torchvision.datasets.CIFAR10(
     #     root='./data', train=False, download=True, transform=transform_test)
-    testset = ImageFolder("dataset/val", transform=transform_test)
+    testset = ImageFolder(f"{args.dataset}/val", transform=transform_test)
     testloader = torch.utils.data.DataLoader(
         testset, batch_size=16, shuffle=False, num_workers=2)
 
@@ -167,7 +175,7 @@ if __name__ == '__main__':
 
     # Model
     print('==> Building model..')
-    net = ResNet18(len(classes))
+    net = ResNet18(len(classes), args.size)
 
     net = net.to(device)
     if device == 'cuda':
@@ -191,15 +199,18 @@ if __name__ == '__main__':
 
     logfile_writer = open(
         f"logs/{datetime.now().strftime('trainlog_%Y%m%d_%H%M%S')}{'_resume' if args.resume else ''}.txt", "w")
+    # logfile_writer.write(
+    #     json.dumps(
+    #         dict(
+    #             learning_rate = args.lr,
+    #             resume=args.resume,
+    #         )
+    #     )+"\n"
+    # )
     logfile_writer.write(
-        json.dumps(
-            dict(
-                learning_rate = args.lr,
-                resume=args.resume,
-            )
-        )+"\n"
+        json.dumps(vars(args))
     )
-    end_epoch = start_epoch+25
+    end_epoch = start_epoch+args.epoch
     for epoch in range(start_epoch, end_epoch):
         train(epoch)
         test(epoch)
diff --git a/models/resnet.py b/models/resnet.py
index 2a420a9..c87f96c 100644
--- a/models/resnet.py
+++ b/models/resnet.py
@@ -71,9 +71,10 @@ class Bottleneck(nn.Module):
 
 
 class ResNet(nn.Module):
-    def __init__(self, block, num_blocks, num_classes=10):
+    def __init__(self, block, num_blocks, num_classes=10, img_size:int=128):
         super(ResNet, self).__init__()
         self.in_planes = 64
+        self.ing_size=img_size
 
         self.conv1 = nn.Conv2d(3, 64, kernel_size=3,
                                stride=1, padding=1, bias=False)
@@ -83,8 +84,10 @@ class ResNet(nn.Module):
         self.layer3 = self._make_layer(block, 256, num_blocks[2], stride=2)
         self.layer4 = self._make_layer(block, 512, num_blocks[3], stride=2)
         # self.linear = nn.Linear(512*block.expansion, num_classes)
-        # self.linear = nn.Linear(8192*block.expansion, num_classes)
-        self.linear = nn.Linear(2048*block.expansion, num_classes)
+        if img_size == 128:
+            self.linear = nn.Linear(8192*block.expansion, num_classes)
+        elif img_size == 64:
+            self.linear = nn.Linear(2048*block.expansion, num_classes)
 
     def _make_layer(self, block, planes, num_blocks, stride):
         strides = [stride] + [1]*(num_blocks-1)
@@ -106,8 +109,8 @@ class ResNet(nn.Module):
         return out
 
 
-def ResNet18(num_classes: int):
-    return ResNet(BasicBlock, [2, 2, 2, 2], num_classes)
+def ResNet18(num_classes: int, img_size:int = 128):
+    return ResNet(BasicBlock, [2, 2, 2, 2], num_classes, img_size)
 
 
 def ResNet34():
